src/main.rs

```diff
@@ -25,75 +25,100 @@ macro_rules! tx {
-    let custom_client = Transaction::new(Arc::new(move |user| {
-        let provider = provider.clone();
-        Box::pin(async move { setup_custom_client(&provider, user).await })
-    }));
-
-    GooseAttack::initialize()?
-        .register_scenario(
-            scenario!("WebsiteUser")
-                // .set_weight(1)?
-                .register_transaction(custom_client.clone().set_name("logon"))
-                // After each transactions runs, sleep randomly from 5 to 15 seconds.
-                .set_wait_time(
-                    Duration::from_secs(wait_time_from),
-                    Duration::from_secs(wait_time_to),
-                )?
-                .register_transaction(tx!(website_index))
-                .register_transaction(tx!(website_openapi))
-                .register_transaction(tx!(website_sboms))
-                .register_transaction(tx!(website_packages))
-                .register_transaction(tx!(website_advisories))
-                .register_transaction(tx!(website_importers)),
-        )
-        .register_scenario(
-            scenario!("RestAPIUser")
-                // .set_weight(1)?
-                .register_transaction(custom_client.clone().set_name("logon"))
-                // After each transactions runs, sleep randomly from 5 to 15 seconds.
-                .set_wait_time(
-                    Duration::from_secs(wait_time_from),
-                    Duration::from_secs(wait_time_to),
-                )?
-                .register_transaction(tx!(list_organizations))
-                .register_transaction(tx!(list_advisory))
-                .register_transaction(tx!(list_advisory_paginated))
-                .register_transaction(tx!(get_advisory_by_doc_id))
-                .register_transaction(tx!(list_vulnerabilities))
-                .register_transaction(tx!(list_vulnerabilities_paginated))
-                .register_transaction(tx!(list_importer))
-                .register_transaction(tx!(list_packages))
-                .register_transaction(tx!(list_packages_paginated))
-                .register_transaction(tx!(search_packages))
-                .register_transaction(tx!(list_products))
-                .register_transaction(tx!(list_sboms))
-                .register_transaction(tx!(list_sboms_paginated)),
-        )
-        .register_scenario(
-            scenario!("GraphQLUser")
-                // .set_weight(1)?
-                .register_transaction(custom_client.set_name("logon"))
-                // After each transactions runs, sleep randomly from 5 to 15 seconds.
-                .set_wait_time(
-                    Duration::from_secs(wait_time_from),
-                    Duration::from_secs(wait_time_to),
-                )?
-                .register_transaction(
-                    tx!(graphql_query_advisory).set_name("query advisory with /graphql"),
-                ),
-        )
-        .execute()
-        .await?;
-
+    if !mem_prof {
```

Comment:
Is there a reason why we need to do this here ?

Reply:
if this is about turning on or off auth, because for some reason it interferes with memory profiling why not switch it there ?

Reply:
thanks for the review
because I'm avoiding the required env_vars 
```
let issuer_url = std::env::var("ISSUER_URL").context("Missing env-var 'ISSUER_URL'")?;
let client_id = std::env::var("CLIENT_ID").context("Missing env-var 'CLIENT_ID'")?;
let client_secret = std::env::var("CLIENT_SECRET").context("Missing env-var 'CLIENT_SECRET'")?;
```

and avoiding to hit this `.register_transaction(custom_client.clone().set_name("logon"))`